### PR TITLE
Fix panic due to nil bind options

### LINKED ISSUE #23852: crash on "service create --mount" when bind-propagation is used
On 1.12.0-rc1-1f136c1 and master-b2da02d:

```
docker service create --name test --mount type=bind,source=/tmp,target=/tmp,writable=true,bind-propagation=shared ubuntu sleep 1d
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x7387bc]

goroutine 1 [running]:
panic(0xc413e0, 0xc8200100b0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/docker/docker/api/client/service.(*MountOpt).Set(0xc820001590, 0x7ffe79097e83, 0x47, 0x0, 0x0)
	/go/src/github.com/docker/docker/api/client/service/opts.go:197 +0x115c
github.com/spf13/pflag.(*FlagSet).setFlag(0xc820074990, 0xc8202fc820, 0x7ffe79097e83, 0x47, 0x7ffe79097e7b, 0x7, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/pflag/flag.go:715 +0x95
github.com/spf13/pflag.(*FlagSet).parseLongArg(0xc820074990, 0x7ffe79097e7b, 0x7, 0xc8203282d0, 0x4, 0x4, 0xc8203282e0, 0x3, 0x3, 0x0, ...)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/pflag/flag.go:776 +0x3ed
github.com/spf13/pflag.(*FlagSet).parseArgs(0xc820074990, 0xc8203282d0, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/docker/docker/vendor/src/github.com/spf13/pflag/flag.go:853 +0x2bd

```

Looks like "mount.BindOptions" is nil

### PR DESCRIPTION
**Pull request by cpuguy83**
Fixes #23852

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM 🐭 assuming it's green :green_heart:

File: api/client/service/opts.go [hunk 1]
```diff
@@ -160,6 +160,13 @@ func (m *MountOpt) Set(value string) error {
 		return mount.VolumeOptions
 	}
 
+	bindOptions := func() *swarm.BindOptions {
```
Comment (LK4D4) [inline api/client/service/opts.go]: what the point of doing this lazily?

Comment (cpuguy83) [inline api/client/service/opts.go]: I just followed what was done for `volumeOptions`, tbh.

Comment (LK4D4) [inline api/client/service/opts.go]: ok :)

Comment (cpuguy83) [inline api/client/service/opts.go]: I guess making sure it stays nil if it it's not needed is a bonus.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**